src/aleph/vm/pool.py

```diff
@@ -239,8 +240,10 @@ async def load_persistent_executions(self):
                 persistent=saved_execution.persistent,
             )
 
-            if execution.is_running:
-                # TODO: Improve the way that we re-create running execution
+            if await self.systemd_manager.is_service_active(
+                execution.controller_service
+            ):  # TODO: Improve the way that we re-create running execution
```

Comment:
IMO having different ways to check if an instance is running here or at the method used on the `VMExecution` class, can provide some issues in the future, because we don't follow always the same pattern to check it.

Reply:
You are correct, but IMO having the same way to check if a VM is running independently of the method used (systemd or direct) it's better that rehydrate fields and trust on that rehydration.